File: abc331/c-problem/src/main.py
import numpy as np
n = int(input())
a = list(map(int, input().split()))
ia_list = list(enumerate(a))
ia_list.sort(key=lambda x: x[1])
ia_list = np.array(ia_list)

b = [0] * n
for i, av in ia_list:
    ia_list = ia_list[ia_list[:, 1] > av, :]
    b[i] = ia_list[:, 1].sum()
print(' '.join(b))

File: abc331/c-problem/src/main.rs
use proconio::input;

fn main() {
    input! {
        n : usize,
        a : [isize; n],
    }
    let mut a_with_i: Vec<(usize, isize)> = a.iter()
        .enumerate()
        .map(|(i, av)| (i, av.clone()))
        .collect();
    a_with_i.sort_by(|(_, av), (_, bv)| av.cmp(bv));
    let mut a_with_i_filtered: Vec<(usize, isize)> = a_with_i.iter().cloned().collect();
    let mut b = vec![0; n];
    for (i, ai) in &a_with_i {
        a_with_i_filtered = a_with_i_filtered.iter().cloned().filter(|(_, av)| av > ai).collect();
        b[*i] = a_with_i_filtered.iter().map(|(_, av)| *av).sum();
    }
    print!(
        "{}",
        b.iter()
            .map(std::string::ToString::to_string)
            .collect::<Vec<_>>()
            .join(" ")
    );
}
